remote/bitbucketserver/bitbucketserver.go

```diff
@@ -4,241 +4,150 @@ package bitbucketserver
 // quality or security standards expected of this project. Please use with caution.
 
 import (
+	"crypto/md5"
 	"crypto/rsa"
+	"crypto/tls"
 	"crypto/x509"
+	"encoding/hex"
 	"encoding/json"
 	"encoding/pem"
 	"fmt"
-	"io/ioutil"
-	"net/http"
-	"net/url"
-
 	log "github.com/Sirupsen/logrus"
```

Comment:
in general we should try to avoid logging at such low levels. Logging is generally at a higher level, for example, in the remote, or when possible, in the handlers. Let's remove logging from this tier